src/platform/plugins/shared/discover/public/application/main/components/top_nav/app_menu_actions/get_share.tsx

```diff
@@ -26,125 +26,151 @@ export const getShareAppMenuItem = ({
-            title: i18n.translate('discover.share.shareModal.title', {
-              defaultMessage: 'Share this Discover session',
-            }),
-            config: {
-              embed: {
-                disabled: true,
-                showPublicUrlSwitch,
-              },
-            },
+    services.share?.toggleShareContextMenu({
+      asExport,
+      anchorElement,
+      allowShortUrl: !!services.capabilities.discover_v2.createShortUrl,
+      shareableUrl,
+      shareableUrlForSavedObject,
+      shareableUrlLocatorParams: { locator, params },
+      objectId: savedSearch.id,
+      objectType: 'search',
+      objectTypeMeta: {
+        title: i18n.translate('discover.share.shareModal.title', {
+          defaultMessage: 'Share this Discover session',
+        }),
+        config: {
+          embed: {
+            disabled: true,
+            showPublicUrlSwitch,
           },
-          sharingData: {
-            isTextBased: isEsqlMode,
-            locatorParams: [
-              {
-                id: locator.id,
-                params: isEsqlMode
-                  ? {
-                      ...params,
-                      timeRange: timefilter.getAbsoluteTime(), // Will be used when generating CSV on server. See `filtersFromLocator`.
-                    }
-                  : params,
-              },
-            ],
-            ...searchSourceSharingData,
-            // CSV reports can be generated without a saved search so we provide a fallback title
-            title:
-              savedSearch.title ||
-              i18n.translate('discover.localMenu.fallbackReportTitle', {
-                defaultMessage: 'Untitled Discover session',
-              }),
-          },
-          isDirty: !savedSearch.id || stateContainer.appState.hasChanged(),
-          onClose: () => {
-            anchorElement?.focus();
+        },
+      },
+      sharingData: {
+        isTextBased: isEsqlMode,
+        locatorParams: [
+          {
+            id: locator.id,
+            params: isEsqlMode
+              ? {
+                  ...params,
+                  timeRange: timefilter.getAbsoluteTime(), // Will be used when generating CSV on server. See `filtersFromLocator`.
+                }
+              : params,
           },
-          toasts: notifications.toasts,
-        });
+        ],
+        ...searchSourceSharingData,
+        // CSV reports can be generated without a saved search so we provide a fallback title
+        title:
+          savedSearch.title ||
+          i18n.translate('discover.localMenu.fallbackReportTitle', {
+            defaultMessage: 'Untitled Discover session',
+          }),
       },
-    },
+      isDirty: !savedSearch.id || stateContainer.appState.hasChanged(),
+      onClose: () => {
+        anchorElement?.focus();
+      },
+      toasts: notifications.toasts,
+    });
   };
+
+  const menuItems: AppMenuActionPrimary[] = [
+    {
+      id: AppMenuActionId.share,
```

Comment:
Showing the absolute time range for the Share modal is not a new UI element, right? While testing this PR I noticed that it might be better if we increased the precision of how it's displayed.

So instead of this format
<img width="557" alt="Screenshot 2025-05-07 at 16 41 49" src="https://github.com/user-attachments/assets/39072405-617b-4bc6-be2c-3014552da52a" />

Something like the format which we show in the sate picker:
<img width="626" alt="Screenshot 2025-05-07 at 16 42 03" src="https://github.com/user-attachments/assets/5e3ec7c3-2cc2-447b-a13c-052274fcc141" />

Just a suggestion to manage users expectations. Wdyt?

Reply:
I think this is a fair suggestion, it's however out of scope for this particular PR